base58/src/error.rs

```diff
@@ -40,73 +40,66 @@ impl From<BadByteError> for DecodeError {
     fn from(e: BadByteError) -> Self { Self::BadByte(e) }
 }
 
-/// An error that might occur during base58 decoding.
+/// An error occurred during base58 decoding (with checksum).
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
-pub enum Error {
+pub enum DecodeCheckError {
     /// Decode error.
     Decode(DecodeError),
-    /// Checksum was not correct (expected, actual).
-    BadChecksum(u32, u32),
-    /// The length (in bytes) of the object was not correct.
-    ///
-    /// Note that if the length is excessively long the provided length may be an estimate (and the
-    /// checksum step may be skipped).
-    InvalidLength(usize),
-    /// Extended Key version byte(s) were not recognized.
-    InvalidExtendedKeyVersion([u8; 4]),
-    /// Address version byte were not recognized.
-    InvalidAddressVersion(u8),
-    /// Checked data was less than 4 bytes.
-    TooShort(usize),
-}
-
-impl fmt::Display for Error {
+    /// Checksum was not correct.
+    BadChecksum(BadChecksumError),
+    /// Checked data was too short.
+    TooShort(TooShortError),
+}
+
+impl fmt::Display for DecodeCheckError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        use Error::*;
+        use DecodeCheckError::*;
 
         match *self {
             Decode(ref e) => write_err!(f, "decode"; e),
-            BadChecksum(exp, actual) =>
-                write!(f, "base58ck checksum {:#x} does not match expected {:#x}", actual, exp),
-            InvalidLength(ell) => write!(f, "length {} invalid for this base58 type", ell),
-            InvalidExtendedKeyVersion(ref v) =>
-                write!(f, "extended key version {:#04x?} is invalid for this base58 type", v),
-            InvalidAddressVersion(ref v) =>
-                write!(f, "address version {} is invalid for this base58 type", v),
-            TooShort(_) => write!(f, "base58ck data not even long enough for a checksum"),
+            BadChecksum(ref e) => write_err!(f, "bad checksum"; e),
```

Comment:
"bad" is a pretty bad word. ;)

Invalid, unexpected, non-matching...

Reply:
"Incorrect" I think is clearest.

Reply:
Sure thing!